core/trino-main/src/main/java/io/trino/operator/project/InputChannels.java

```diff
@@ -51,6 +51,11 @@ public Page getInputChannels(Page page)
         return page.getColumns(inputChannels);
     }
 
+    public Page getLoadedInputChannels(Page page)
```

Comment:
You don't know if all blocks will be loaded since expressions can be conditional (e.g. `CASE WHEN`). We already load block for single channel `PageProjection`

Reply:
Right, missed that
The current trade off of going via lazy block wrapper for every position just to preserve laziness isn't great but this is probably better handled through vectorized rewrite in future.